### ISSUE TITLE
Training data and testing data for Caffe Model

### ISSUE THREAD

**Issue by vinitveersingh** [nealjean/predicting-poverty#27]
Greetings Authors,
I have a few questions on the training data set for the Caffe Model. After going over your code, it seems like all modified coordinates from LSMS were used in creating the downloaded_locs.txt, and downloaded_locs.txt is used by extract_features.py and thus becomes the test set before it could be used regression. To simplify, I am concerned if training set and testing set were mutually exclusive? If yes is it possible for you to share the training coordinates for each country?

Thanks,
Vinit

**Comment by nealjean** [comment]
Hi Vinit,

The Caffe model was trained on a separate dataset of satellite images and corresponding nighttime light labels - this training dataset was sampled from all over Africa near DHS survey locations and is described in [this paper](https://arxiv.org/abs/1510.00098).

Hope that is helpful!

**Comment by vinitveersingh** [comment]
Hi Neal,
Thank you for replying. I read through the paper and from what I understand, the  330,000 images training images came by randomly sampling near DHS survey locations. And to illustrate this you mark these locations as white. However, due to an imbalance in the dataset w.r.t night light intensity, you create a 3-class balanced dataset consists of 150,000 training images ( 50k per class) and 8,000
validation images.
And from the LSMS the survey, for a given group, you sample approximately 100 1km×1km images tiling a 10km × 10km area centered at the average household location as input. 
             
So, to summarize,  
**Training set** : 
Size: {Xi, Yi} x 150,000
 Xi : is the daytime image taken **near a DHS survey location** 
 Yi :  night light intensity bin corresponding to the same location as Xi

** Validation set**
Size: {Xi, Yi} x 8,000
 Xi : is the daytime image taken **near a DHS survey location** 
 Yi :  night light intensity bin corresponding to the same location as Xi

**Test set for ** 
Size: {Xi} x (Number of [image_lat] and [image_lon] coordinate in downloaded_locs.txt genrated from LSMS for a given country 
Xi : is the daytime image taken at **[image_lat] and [image_lon] at zoom level 16** 
Yi: Nightlight bin predicted by the network

Is my understanding here correct? And if yes, while I understand it might be reasonable to assume that locations of the DHS training images and LSMS test images are not the overlapping much since they are taken from two different sources, what I cannot understand is how is the location for calculating DHS asset score are different from the training set. And what happens to remaining 172000 (  330,000 - 150,000 - 8000) images from your training set. Do you simply discard these images?

Again, if possible and available it would be extremely helpful if you could share the training locations since I am trying to replicate your experiments with a different network and I don't want different locations to be a nuisance factor while comparing my result with yours.  

Thank you,
Vinit

**Comment by nealjean** [comment]
@sangmichaelxie do you have the training locations for the nightlights task?

@vinitveersingh I think there is a slight misunderstanding for the test set. After we train the CNN to predict nighttime lights, we use it as a fixed feature extractor for the poverty prediction task. What that means is that for each of the LSMS locations, we sample ~100 images from the 10x10 square and then run those through the CNN to get the 100 corresponding feature vectors. We then average those feature vectors to get one feature vector for each LSMS (or DHS) cluster, and use those features for the ridge regression model.

**Comment by vinitveersingh** [comment]
Hi Neal,
Thanks for the correction. And thanks for asking @sangmichaelxie for the training locations.

**Comment by vinitveersingh** [comment]
Good Afternoon Neal,
Were you or Michael able to find the file containing the locations you used for training your network? Will it be possible to share it?

Thank you,
Vinit

**Comment by nealjean** [comment]
I don't have the file

**Comment by sangmichaelxie** [comment]
Hi Vinit, 

We don't prefer to share the training locations. We've actually tried many different architectures and different samples of training sets - the method is robust to this. Note that the Google Static Map images have also updated since our 2016 paper, so even if you use the same locations, the images will not be exactly the same. So the original locations wouldn't result in the same dataset.

**Comment by vinitveersingh** [comment]
Neal and Michael thanks for answering my questions.
Sincerely,
Vinit
